project-1_test/main.py

```diff
@@ -0,0 +1,151 @@
+        raise ValueError("The sheet name expected a string object")
+
+
+def create_spreadsheet(sheet_name):
+    """
+    This creates a new spreadsheet file in google sheet
+    """
+    if isinstance(sheet_name, str):
+        sh = gc.create(sheet_name)
+        logging.info("Spreadsheet created")
+        return sh
+    else:
+        raise ValueError("The sheet name expected a string object")
+
+
+def select_spreadsheet(filename, sheet_name):
+    """
+    This function  selects the preferred worksheet
+    :param filename: This is the name of the spreadsheet
+    :param sheet_name: This is the name of the desired worksheet in str format
+    """
+    if isinstance(filename, str) and isinstance(sheet_name, str):
+        worksheets = open_spreadsheet(filename).worksheet(sheet_name)
+        logging.info(f"Worksheet {sheet_name} has been loaded")
+        return worksheets
+    else:
+        raise ValueError("The input expected a string object")
+
+
+def spreadsheet_format(filename, sheet_name, first_column, second_column):
+    """
+    This functon is for the formatting of the spreadsheet
+    :param sheet_name: Name of current worksheet
+    :param filename: Name of spreadsheet
+    :param first_column: This is the name of the start column in the range in str format
+    :param second_column: This is the name of the end column in the range in str format
+    """
+    if (isinstance(filename, str) and isinstance(sheet_name, str)
+            and isinstance(first_column, str) and isinstance(second_column, str)):
+        formats = (select_spreadsheet(filename, sheet_name).format
+                   (f"{first_column}:{second_column}", {'textFormat': {'bold': True}}))
+        logging.info("Worksheet columns format has been set")
+        return formats
+    else:
+        raise ValueError("The input expected a string object e.g 'A1' for the cols and rows input."
+                         "Wrong input expected a string object.")
+
+
+def new_worksheet(filename, new_name, rows, columns):
+    """
+    This function creates a new worksheet within the spreadsheet.
+    :param filename: This is the name of the spreadsheet file
+    :param new_name: Name of  the new worksheet
+    :param rows: Number of desired rows
+    :param columns: Number of desired columns
+    """
+
+    if (isinstance(rows, int) and isinstance(columns, int) and
+            isinstance(filename, str) and isinstance(new_name, str)):
+        open_s = open_spreadsheet(filename)
+        sheet_names = [s.title for s in open_s.worksheets()]
+        if new_name not in sheet_names:
+            open_spreadsheet(filename).add_worksheet(new_name, rows, columns)
+            logging.info("Worksheet created")
+        else:
+            raise FileExistsError("This worksheet already exists")
+    else:
+        raise ValueError("The input expected a number/integer for the "
+                         "rows ands cols input, other inputs are strings")
+
+
+driver = chrome_driver()
```

Comment:
you might want to refactor your code to use 

```
if __main__ == "__name__"
```

because you don't want your whole code to run when you run your tests

Reply:
I have done that.